src/backend/rest.ts

```diff
@@ -58,13 +59,100 @@ type RestApiFilterRequest = {
   filter: unknown;
 } & CommonRequest;
 
+type RestApiSiteConnectionResponse = {
+  value: Array<{
+    extensions: {
+      basic_settings: {
+        alias: string;
+        site_id: string;
+      };
+    };
+  }>;
+};
+
+type RestApiHostResponse = {
+  value: Array<{
+    id: string;
+    extensions: {
+      name: string;
+    };
+  }>;
+};
+
+type RestApiServiceResponse = {
+  value: Array<{
+    id: string;
+    extensions: {
+      description: string;
+    };
+  }>;
+};
+
 export default class RestApiBackend implements Backend {
   datasource: DatasourceOptions;
 
   constructor(datasource: DatasourceOptions) {
     this.datasource = datasource;
   }
 
+  async filterSites(): Promise<MetricFindValue[]> {
+    const query = {};
+    const response = await this.api<RestApiSiteConnectionResponse>({
+      url: '/domain-types/site_connection/collections/all',
+      method: 'GET',
+      params: {
+        query: JSON.stringify(query),
+      },
+    });
+    return response.data.value.map((element) => {
+      const bs = element.extensions.basic_settings;
+      return { text: bs.alias, value: bs.site_id, expandable: false };
+    });
+  }
+
+  async filterHosts(query: MetricFindQuery): Promise<MetricFindValue[]> {
+    const liveStatusQuery = toLiveStatusQuery(replaceVariables(query.filter), 'host');
+    const response = await this.api<RestApiHostResponse>({
+      url: '/domain-types/host/collections/all',
+      method: 'GET',
+      params: {
+        query: JSON.stringify(liveStatusQuery.query),
+        sites: liveStatusQuery.sites,
+        columns: 'name',
+      },
+    });
+    return response.data.value.map((element) => {
+      return { text: element.extensions.name, value: element.id, expandable: false };
+    });
+  }
+
+  async filterServices(query: MetricFindQuery): Promise<MetricFindValue[]> {
+    const liveStatusQuery = toLiveStatusQuery(replaceVariables(query.filter), 'service');
+    const response = await this.api<RestApiServiceResponse>({
+      url: '/domain-types/service/collections/all',
+      method: 'GET',
+      params: {
+        query: JSON.stringify(liveStatusQuery.query),
+        sites: liveStatusQuery.sites,
+        columns: 'description',
+      },
+    });
+    return response.data.value.map((element) => {
+      return { text: element.extensions.description, value: element.extensions.description, expandable: false };
+    });
+  }
```

Comment:
These three could maybe be implemented by calling a common utility method, I think. Their structure is basically the same.

I'm also not sure about the names, at least the first one. It doesn't do any filtering, doesn't it?

Reply:
yes, i also thought about combining them, but my final thought was that the differences are big enought that the resulting common function is more complicate than the duplicated independent one.
i would rename them, but keep them as is. if you have objection feel free to reopen the discussion!